rfcs/0171-fetch-from-github.md

```diff
@@ -0,0 +1,114 @@
+---
+feature: Default name of fetchFromGithub FOD to include revision
+start-date: 2024-03-15
+author: Jonathan Ringer
+co-authors: 
+shepherd-team:  
+shepherd-leader: 
+related-issues:
+---
+
+# Summary
+[summary]: #summary
+
+Updating the hash on Fixed-Output Derivations (FODs) is a very error prone process. It's not intuitive to invalidate the existing hash, attempt to realize the FOD, then replace the hash value with the value that Nix just calculated. This RFC advocates for influencing the default derivation name of the fetchFromGithub helper with the "repo" and "rev" values to ensure that changed URLs force the FOD to be re-built.
+
+# Motivation
+[motivation]: #motivation
+
+This will hopefully provide immediate feedback that an FOD contains a stale hash. One must either build the FOD without the previous FOD in their Nix store, or run the FOD build with `--check` to verify that the FOD is not stale. Although fetchFromGithub is one of many fetchers; it is very common, and generally has a user specify the source information in granular which makes intent clear.
```

Comment:
```suggestion
This will hopefully provide immediate feedback that an FOD contains a stale hash. One must either build the FOD without the previous FOD in their Nix store, or run the FOD build with `--check` to verify that the FOD is not stale. Although fetchFromGithub is one of many fetchers; it is very common, and generally has a user specify granular source information which makes differentiating between sources easy.
```